src/stm32f1/adc.cpp

```diff
@@ -164,10 +285,10 @@ float adc::driver_read()
          0) {
   }
 
-  constexpr auto full_scale_max = bit_limits<12, size_t>::max();
-  constexpr auto full_scale_float = static_cast<float>(full_scale_max);
+  auto const full_scale_max = bit_limits<12, size_t>::max();
```

Comment:
```suggestion
  constexpr auto full_scale_max = bit_limits<12, size_t>::max();
```

If something can be constexpr make it constexpr. It was correct last time :)